##7.1异常的体系  
Throwable 所有异常或者错误类的超类.  
Error     错误,错误一般是jvm或者是硬件引发的问题,一半不会去通过代码去处理错误.  
Exception 异常,需要通过代码去处理.  

    public class l {
        public static void main(String arg[]) {
            Throwable t = new Throwable("头疼");    //创建一个Throwabl类型对象
            String tostr = t.toString();          
            System.out.println("tostring:" + tostr);
            String get = t.getMessage();
            System.out.println("getMessage:" + get);
            t.printStackTrace();    
            String loc = t.getLocalizedMessage();
            System.out.println("getLocalizedMessage:" + loc);
        }
    }
输出:
tostring:java.lang.Throwable: 头疼  
getMessage:头疼  
java.lang.Throwable: 头疼  
	at l.main(l.java:13)  
getLocalizedMessage:头疼  

Throwable 常用的方法:  
1. toString() 返回当前一场对象的完整类名  
2. getMessage()返回的是创建throwable传入的字符串信息  
3. printStackTrace() 打印异常的栈信息  
***
##7.2处理异常
###7.2.1使用try..catch处理异常


捕获处理要注意的细节:  
1. 如果try块中代码除了异常经过了处理之后,那么try-catch块外面的代码可以正常执行.  
2. 如果try块中出了异常的代码,那么在try块中出现异常代码后面的代码是不会执行了.  
3. 一个try块后面是可以跟有多个catch块的,也就是一个try块可以捕获多种异常的类型.  
4. 一个try块可以捕获多种异常的类型,但是捕获的异常类型必须从小到大进行捕获,否则编译报错.  


